core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -1664,7 +1700,20 @@ private ContextImpl(DatabaseProduct databaseProduct,
           databaseVersion, databaseMajorVersion, databaseMinorVersion,
           literalQuoteString, literalEscapedQuoteString,
           identifierQuoteString, identifierEscapedQuoteString,
-          quotedCasing, unquotedCasing, caseSensitive,
+          charset, quotedCasing, unquotedCasing, caseSensitive,
+          conformance, nullCollation, dataTypeSystem, jethroInfo);
+    }
+
+    @Override public String charset() {
+      return charset;
+    }
+
+    @Override public Context withCharset(String charset) {
```

Comment:
does this qualify as a breaking change which has to be documented in history.md?

Reply:
I don't think so. ContextImpl is private.